containerd-shim-v2/service.go

```diff
@@ -56,6 +57,10 @@ var vci vc.VC = &vc.VCImpl{}
 // New returns a new shim service that can be used via GRPC
 func New(ctx context.Context, id string, publisher events.Publisher) (cdshim.Shim, error) {
 	logger := logrus.WithField("ID", id)
+	// Discard the log before shim init its log output. Otherwise
+	// it will output into stdio, from which containerd would like
+	// to get the shim's socket address.
+	logrus.SetOutput(ioutil.Discard)
```

Comment:
Ouch - yes I've seen problems like this before. But won't we lose potentially useful information here? Is there a way to suppress the log output until after containerd gets the socket address? Alternatively, can we redirect this log output somewhere else?

Reply:
Hi @jodh-intel 
So how about passing "ignoreLogging=false" to LoadConfiguration() in which will add hook logging into syslog?

Reply:
Sounds good. That would make the shim work more like the current runtime. It would also mean we can make use of https://github.com/kata-containers/tests/tree/master/cmd/log-parser for the shim.